src/checkActivities.rb

```diff
@@ -0,0 +1,136 @@
+module ActivityCheck
+
+    def self.initialize(activityDirName)
+        @activityDirName = activityDirName
+
+        # create js file name array
+        @jsFilesArray = Dir.glob("../src/activities/" + @activityDirName + "/**/*.js")
+
+        # create qml file name array
+        @qmlFilesArray = Dir.glob("../src/activities/" + @activityDirName + "/**/*.qml")
+
+    end
+
+    # to print error
+    def self.print_error(err)
+        puts "\t ERROR       #{err}"
+    end
+
+    # to print warning
+    def self.print_warning(wrn)
+        puts "\t WARNING     #{wrn}"
+    end
+
+    # checks import version
+    def self.check_import_version(lineStr)
+        # add modules and versions in hash
+        moduleVersionHash = { "QtQuick" => "2.6", "GCompris" => "1.0" }
+
+        matchStr = /import (?<module>(\w+)) (?<ver>(\d+.\d+))/.match(lineStr)
+        if matchStr
+            moduleName = matchStr["module"]
+            correctModuleVersion = moduleVersionHash[moduleName]
+            if matchStr["ver"] != correctModuleVersion
+                print_error "#{moduleName} version must be #{correctModuleVersion}"
+            end
+        end
+    end
+
+    # checks credit updation
+    def self.check_credits_update(lineStr)
+        if lineStr.include?("THE GTK VERSION AUTHOR")
+            print_error "Replace original GTK VERSION AUTHOR label by your own name"
+        end
+
+        if lineStr.include?("THE GTK VERSION AUTHOR")
+            print_warning "Replace original QT Quick copyright label email by your own email address"
+        end
+    end
+
+    # checks whether qsTr can be used in the lineStr
+    def self.qsTr_use(lineStr, lineNumber)
+        matchStr = /([^ ]+): "[^"]+"/.match(lineStr)
+        if matchStr
+            print_warning "line:#{lineNumber} #{matchStr} qsTr may not be used"
+        end
+    end
+
+    # checks copyrights updated
+    def self.activityInfo_QmlFile(lineStr, directoryName)
+        valuesToTestArray = ["author: \"Your Name &lt;yy@zz.org&gt;\"", "name: \"test/Test.qml\"", "difficulty: 1", "icon: \"test/test.svg\"",

```

Comment:
for "demo: false", can you only display the warning if Bruno Coudoin is not the author?

Reply:
i don't understand. Why is that so ? every activity he developed does not have a demo ?

Reply:
Only the activities developed by Bruno should have demo: false